transformer_engine/pytorch/attention.py

```diff
@@ -3596,26 +3628,17 @@ def forward(
             batch_start = inference_params.batch_size_offset
             batch_end = batch_start + key_layer.size(1)
             assert batch_end <= inference_key_memory.size(1)
+
             sequence_start = inference_params.sequence_len_offset
             sequence_end = sequence_start + key_layer.size(0)
             assert sequence_end <= inference_key_memory.size(0)
```

Comment:
We don't need all this logic here since KV cache logic is handled inside DPA:
```suggestion
            sequence_start = inference_params.sequence_len_offset
            sequence_end = sequence_start + key_layer.size(0)
```